ratatool-sampling/src/main/scala/com/spotify/ratatool/samplers/BigSampler.scala

```diff
@@ -412,6 +555,43 @@ private[samplers] object BigSamplerBigQuery {
   }
   // scalastyle:on cyclomatic.complexity
 
+  // scalastyle:off cyclomatic.complexity
+  // TODO: Potentially reduce this and hashAvroField to a single function
+  @tailrec
+  private[samplers] def getTableRowField(r: TableRow,
+                                         fieldStr: String,
+                                         tblSchema: Seq[TableFieldSchema]): Any = {
+    val subfields = fieldStr.split(BigSampler.fieldSep)
+    val field = tblSchema.find(_.getName == subfields.head).getOrElse {
+      throw new NoSuchElementException(s"Can't find field `$fieldStr` in the schema $tblSchema")
+    }
+    val v = r.get(subfields.head)
+    if (v == null) {
+      log.debug(s"Field `${field.getName}` of type ${field.getType} and mode ${field.getMode}" +
+        s" is null - won't account for hash")
+    } else {
+      field.getType match {
+        case "BOOLEAN" => v.toString.toBoolean
+        case "INTEGER" => v.toString.toLong
+        case "FLOAT" => v.toString.toFloat
+        case "STRING" => v.toString
+        case "BYTES" => BaseEncoding.base64().decode(v.toString)
+        case "TIMESTAMP" => v.toString
+        case "DATE" => v.toString
+        case "TIME" => v.toString
+        case "DATETIME" => v.toString
+        case "RECORD" if fieldStr.nonEmpty =>
+          getTableRowField(
+            v.asInstanceOf[TableRow],
+            subfields.tail.mkString(BigSampler.fieldSep.toString),
+            field.getFields.asScala)
+        case t => throw new UnsupportedOperationException(
```

Comment:
similarly probably add a `todo` for `REPEATED` fields

also does this currently allow `NULLABLE` field types